### DISCUSSION TITLE
How to perform projections on types other than `IQueryable<T>`?

### DISCUSSION THREAD

**Discussion by arjundnair-fingent** [ChilliCream/hotchocolate#5104] [Q&A]
Hi,

I read from the HotChocolate documentation that projections work on `IQueryable<T>` by default. I would like to know whether it is possible to perform projections on types other than `IQueryable<T>` like `Enumerable<T>` or `List<T>`. If it is possible, please do explain how to do it with an example. I've searched a lot on the internet for a solution, but not able to find anything related to this.

**Comment by PascalSenn** [comment]
In V13 we provide a IProjectionContext that you can use to project yourself. Otherwise there is [UseFirstOrDefault] or [UseSingleOrDefault].

**Comment by denis-trtnr** [comment]
Hi @arjundnair-fingent ,

I had the same problem because I needed the response as a List<>. This is how it worked for me while using the Entity Framework:

Here is my query:
`[UseProjection]
        [UseFiltering]
        [UseSorting]
        public async Task<IEnumerable<TechData>> GetTechDataV2([Service] TechDataRepository techDataRepository, IResolverContext rcontext, string matNr, CancellationToken cancellationToken)
            => await techDataRepository.GetAllTechDataByMatNr(rcontext, matNr, cancellationToken);`

And here my service
`public async Task<IEnumerable<TechData>> GetAllTechDataByMatNr(IResolverContext resolverContext, string matNr, CancellationToken cancellationToken = default)
        {
            return _context.TechData.AsQueryable().Project(resolverContext).Where(t => t.MaterialId == matNr).ToList();
        }`
